test/cloud-metadata/_lib.js

```diff
@@ -243,6 +243,60 @@ function loadFixtureData(provider, fixtureName) {
   return fixture;
 }
 
+/**
+ * this function returns a HTTP server which replaces the previous usage of `express`
+ * for mocking cloud metadata responses.
```

Comment:
nit:

```suggestion
 * This function returns a HTTP server similar enough to Express for these tests,
 * but avoids using Express because v5 only works with node >=18.
```